src/test_reference.jl

```diff
@@ -152,33 +150,44 @@ function test_reference(
         @test true # to increase test counter if reached
     else  # When test fails
         # Saving actual file so user can look at it
-        actual_path = joinpath(mismatch_staging_dir(), reference_filename)
-        actual_file = typeof(reference_file)(actual_path)
-        savefile(actual_file, actual)
+        actual_path = save_mismatch_file(reference_file, actual)
 
         # Report to user.
         @info """Reference Test for \"$reference_filename\" failed:
         $(render(rendermode, reference, actual))
-        """ reference = reference_path actual = actual_path
-        
-        if !isinteractive() && !force_update()
-            error("""
-            To update the reference images either run the tests interactively with 'include(\"test/runtests.jl\")',
-            or to force-update all failing reference images set the environment variable `JULIA_REFERENCETESTS_UPDATE`
-            to "true" and re-run the tests via Pkg.
-            """)
-        end
-
-        if force_update() || input_bool("Replace reference with actual result?")
-            mv(actual_path, reference_path; force=true)  # overwrite old file it
-            @info "Please run the tests again for any changes to take effect"
-        else
-            @test false
-        end
+        """ reference=reference_file.filename actual=actual_path
+        handle_mismatch(reference_file, actual_path)
+    end
+end
+
+function save_mismatch_file(reference_file::File{F}, actual) where F
+    # Saving actual file so user can look at it
+    actual_path = joinpath(mismatch_staging_dir(), basename(reference_file.filename))
+    actual_file = File{F}(actual_path)
+    
+    savefile(actual_file, actual)
+    return actual_path
+end
+
+function handle_mismatch(reference_file, actual_path)
+    if !isinteractive() && !force_update()
+        error("""
+        To update the reference images either run the tests interactively with 'include(\"test/runtests.jl\")',
+        or to force-update/create all failing reference images set the environment variable `JULIA_REFERENCETESTS_UPDATE`
+        to "true" and re-run all tests.
+        """)
+    end
+
+    if force_update() || input_bool("Replace reference with actual result?")
+        mkpath(dirname(reference_file.filename))
+        mv(actual_path, reference_file.filename; force=true)  # overwrite old file
+        @info "Please run the tests again for any changes to take effect"
+    else
+        @test false
     end
 end
 
-force_update() = tryparse(Bool, get(ENV, "JULIA_REFERENCETESTS_UPDATE", "false")) === true
+force_update() = tryparse(Bool, get(ENV, "JULIA_REFERENCETESTS_UPDATE", "false"))
```

Comment:
Seems problematic as this function can now return `nothing` which will fail when it's used in a conditional. I do think it's reasonable to throw an exception if a user sets this incorrectly though:
```suggestion
force_update() = parse(Bool, get(ENV, "JULIA_REFERENCETESTS_UPDATE", "false"))
```
Alternatively, we can just silently default to `false`:
```suggestion
force_update() = get(ENV, "JULIA_REFERENCETESTS_UPDATE", "false") == "true"
```